src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingSubmissionIntegrationTest.java

```diff
@@ -765,9 +765,10 @@ void testGetProgrammingSubmissionWithoutAssessmentWithIndividualDueDate(boolean
         }
     }
 
-    @Test
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
     @WithMockUser(username = "tutor1", roles = "TA")
-    void getProgrammingSubmissionWithoutAssessmentAlreadyAssessedNoFound() throws Exception {
+    void test_getProgrammingSubmissionWithoutAssessmentNothingAvailable(boolean lock) throws Exception {
```

Comment:
This was the name before and @JohannesStoehr suggested the give it the prefix test. Do we have a general guideline here? I think I saw many methods with test in front as well in the codebase.

Reply:
```suggestion
    void testGetProgrammingSubmissionWithoutAssessmentNothingAvailable(boolean lock) throws Exception {
```
The `test` prefix is used in most test classes, I think. Some classes do not use it probably by preference of the original author who introduced it. It seems like in this case it isn’t even consistent within the class. AFAIK, there is no guideline for it.

The `test_` prefix is only used in the integration tests that are shared for GitLab/Jenkins and Bamboo/Bitbucket so that the `test_X` method contains the actual assertions. Then for both setup variants a separate `testX` method with the actual `@Test` annotation exists that does some mocking around it but otherwise just calls the generic part.
I don’t think it should be used anywhere else, as underscores in methods names are against the usual Java naming convention.